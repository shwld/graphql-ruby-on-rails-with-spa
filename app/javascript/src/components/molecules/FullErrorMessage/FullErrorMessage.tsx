import { ApolloError } from '@apollo/client/errors'
import React from 'react'

export const FullErrorMessage: React.FC<{
  error?: ApolloError
}> = ({ error }) => {
  if (!error || !error.graphQLErrors || error.graphQLErrors.length === 0) {
    return <></>
  }
  const messages = error.graphQLErrors
    .filter((it) => it.extensions?.code === 'RECORD_INVALID')
    .map((it) => it.message.split(','))
    .flat()
  /** HACK: Railsのエラーオブジェクトそのまま使っているが、ちょっと構造が複雑でが苦しい感じある */
  return (
    <>
      {messages.map((message, i) => (
        <p key={i} className="text-red-500 text-xs italic">
          {message}
        </p>
      ))}
    </>
  )
}
