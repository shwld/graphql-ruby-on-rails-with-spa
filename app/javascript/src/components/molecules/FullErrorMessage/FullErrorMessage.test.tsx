import React from 'react'
import { create } from 'react-test-renderer'

import { FullErrorMessage } from './FullErrorMessage'

describe('Molecules/FullErrorMessage', () => {
  it('Snap Shot', () => {
    const component = create(<FullErrorMessage />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
