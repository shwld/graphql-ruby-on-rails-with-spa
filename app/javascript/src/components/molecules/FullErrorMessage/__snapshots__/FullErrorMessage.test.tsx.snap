// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/FullErrorMessage Snap Shot 1`] = `null`;
